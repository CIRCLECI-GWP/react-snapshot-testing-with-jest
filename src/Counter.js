import React, { useState } from "react";

const Counter = () => {
  const [count, setCount] = useState(0);

  const increment = () => {
    setCount(count + 1);
  };

  const decrement = () => {
    setCount(Math.max(0, count - 1));
  };

  return (
    <div className="counter">
      <h1>COUNTER</h1>
      <div className="buttons">
        <button onClick={increment}>Increment</button>
        <button onClick={decrement}>Decrement</button>
      </div>
      <p>{count}</p>
    </div>
  );
};

export default Counter;
