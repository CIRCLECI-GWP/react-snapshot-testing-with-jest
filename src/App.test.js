import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import Counter from "./Counter";

describe("Jest Snapshot testing suite", () => {
  it("Matches DOM Snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render 3 after three increments", () => {
    render(<Counter />);

    const incrementButton = screen.getByRole("button", { name: /increment/i });

    fireEvent.click(incrementButton);
    fireEvent.click(incrementButton);
    fireEvent.click(incrementButton);

    expect(screen.getByText("3")).toMatchSnapshot();
  });
});
