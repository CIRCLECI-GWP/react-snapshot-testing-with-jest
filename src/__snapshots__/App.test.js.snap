// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="counter"
    >
      <h1>
        COUNTER
      </h1>
      <div
        class="buttons"
      >
        <button>
          Increment
        </button>
        <button>
          Decrement
        </button>
      </div>
      <p>
        0
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Jest Snapshot testing suite Should render 3 after three increments 1`] = `
<p>
  3
</p>
`;
